crates/factors-executor/src/lib.rs

```diff
@@ -0,0 +1,167 @@
+use std::collections::HashMap;
+
+use anyhow::Context;
+use spin_app::{App, AppComponent};
+use spin_core::Component;
+use spin_factors::{AsInstanceState, ConfiguredApp, RuntimeFactors, RuntimeFactorsInstanceState};
+
+/// A FactorsExecutor manages execution of a Spin app.
+///
+/// `Factors` is the executor's [`RuntimeFactors`]. `ExecutorInstanceState`
+/// holds any other per-instance state needed by the caller.
+pub struct FactorsExecutor<Factors: RuntimeFactors, ExecutorInstanceState = ()> {
+    factors: Factors,
+    core_engine: spin_core::Engine<InstanceState<Factors::InstanceState, ExecutorInstanceState>>,
+    configured_app: ConfiguredApp<Factors>,
+    component_instance_pres: HashMap<String, InstancePre<Factors, ExecutorInstanceState>>,
+}
+
+/// A ComponentLoader is responsible for loading Wasmtime [`Component`]s.
+pub trait ComponentLoader {
+    /// Loads a [`Component`] for the given [`AppComponent`].
+    fn load_component(&mut self, component: &AppComponent) -> anyhow::Result<Component>;
+}
+
+type InstancePre<Factors, ExecutorInstanceState> = spin_core::InstancePre<
+    InstanceState<<Factors as RuntimeFactors>::InstanceState, ExecutorInstanceState>,
+>;
+
+impl<Factors: RuntimeFactors, ExecutorInstanceState: Send + 'static>
```

Comment:
Nit: I very much prefer short generic names like `F`. I believe we've had a discussion about this before and have differing opinions, but I personally find the more verbose names trick me into thinking that there is a concrete type called `Factors` instead of reminding me that we're dealing with a generic.

Reply:
I mostly find it confusing when _writing_ code but yeah I'm fine shortening these.